main.nf

```diff
@@ -2077,6 +2088,58 @@ if ( params.gatk_ug_jar != '' ) {
   """
  }
 
+ // pileupCaller for 'random sampling' genotyping
+
+if (params.pileupcaller_bedfile.isEmpty()) {
+  ch_bed_for_pileupcaller = file('NO_FILE_BED')
+} else {
+  ch_bed_for_pileupcaller = Channel.fromPath(params.pileupcaller_bedfile)
+}
+
+if (params.pileupcaller_snpfile.isEmpty ()) {
+  ch_snp_for_pileupcaller = file('NO_FILE')
+} else {
+  ch_snp_for_pileupcaller = Channel.fromPath(params.pileupcaller_snpfile)
+}
+
+
+
+ process genotyping_pileupcaller {
+  label 'mc_small'
+  tag "${samplename}"
+  publishDir "${params.outdir}/genotyping", mode: 'copy'
+
+  when:
+  params.run_genotyping && params.genotyping_tool == 'pileupcaller'
+
+  input:
+  tuple samplename, libraryid, lane, seqtype, organism, strandedness, udg, file(bam), file(bai) from ch_damagemanipulation_for_genotyping_pileupcaller
+  file fasta from ch_fasta_for_genotyping_pileupcaller.collect()
+  file fai from ch_fai_for_pileupcaller.collect()
+  file dict from ch_dict_for_pileupcaller.collect()
+  file bed from ch_bed_for_pileupcaller
+  file snp from ch_snp_for_pileupcaller
+
+  output:
+  tuple samplename, libraryid, lane, seqtype, organism, strandedness, udg, file("pileupcaller.*")
+
+  script:
+  caller = "--${params.pileupcaller_caller}"
+  if (params.pileupcaller_majority_call) {
+    caller = "--majorityCall"
+  }
+  if (params.pileupcaller_random_diploid) {
+    caller = "--randomDiploid"
+  }
+  ssmode = ""
+  if (params.pileupcaller_single_stranded || strandedness == 'single' ) {
```

Comment:
You don't need `params.pileupcaller_single_stranded` in this condition, because supplying the input as direct FASTQ files (rather than TSV), we make a 'fake' TSV-like map from all input flags.